### ISSUE TITLE
Tuya Mmwave Radar Zigbee 3.0

### ISSUE THREAD

**Issue by pinpoint90** [athombv/com.tuya#273]
### Homey version

Homey Pro 12.1.0

### Tuya app version

1.4.0

### Description

Device in homey does not give information about presence detected. The device settings displays that everything is ok but the device cannot be manually controlled. (it is added as presense sensor device and not other) It is also very complicated when adding flows. I would prefer to "if presence detector is activated/deactivated" do action.
Instead I only have the following options "status code gets a new bootlean value/JSION value/Number value/String value". 
The old Tuya Cloud 

### Device specification

```json
{
  "device": {
    "active_time": 1711464028,
    "biz_type": 0,
    "category": "hps",
    "create_time": 1711464028,
    "icon": "smart/icon/ay1531985537002uUUZI/3c9ccef5a17bde28bf29912d581af4f2.jpg",
    "id": "bf58a21ad583637048futo",
    "ip": "<redacted>",
    "lat": "<redacted>",
    "local_key": "<redacted>",
    "lon": "<redacted>",
    "model": "ZG-205Z",
    "name": "Stua: Sensor",
    "node_id": "a4c13815e6331536",
    "online": true,
    "owner_id": "<redacted>",
    "product_id": "2aaelwxk",
    "product_name": "Human presence sensor",
    "status": [
      {
        "code": "presence_state",
        "value": "none"
      },
      {
        "code": "sensitivity",
        "value": 6
      },
      {
        "code": "near_detection",
        "value": 0
      },
      {
        "code": "far_detection",
        "value": 500
      },
      {
        "code": "human_motion_state",
        "value": "none"
      },
      {
        "code": "presence_time",
        "value": 120
      },
      {
        "code": "motionless_far_detection",
        "value": 500
      },
      {
        "code": "motionless_sensitivity",
        "value": 7
      },
      {
        "code": "illuminance_value",
        "value": 0
      },
      {
        "code": "indicator",
        "value": true
      }
    ],
    "sub": true,
    "time_zone": "+01:00",
    "uid": "<redacted>",
    "update_time": 1728443324,
    "uuid": "<redacted>"
  },
  "specifications": {
    "category": "hps",
    "functions": [
      {
        "code": "sensitivity",
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "near_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":1000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "far_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":1000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "presence_time",
        "type": "Integer",
        "values": "{\"unit\":\"s\",\"min\":0,\"max\":28800,\"scale\":0,\"step\":1}"
      },
      {
        "code": "motionless_far_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":1}"
      },
      {
        "code": "motionless_sensitivity",
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "indicator",
        "type": "Boolean",
        "values": "{}"
      }
    ],
    "status": [
      {
        "code": "presence_state",
        "type": "Enum",
        "values": "{\"range\":[\"none\",\"presence\"]}"
      },
      {
        "code": "sensitivity",
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "near_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":1000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "far_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":1000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "human_motion_state",
        "type": "Enum",
        "values": "{\"range\":[\"none\",\"large_move\",\"small_move\"]}"
      },
      {
        "code": "presence_time",
        "type": "Integer",
        "values": "{\"unit\":\"s\",\"min\":0,\"max\":28800,\"scale\":0,\"step\":1}"
      },
      {
        "code": "motionless_far_detection",
        "type": "Integer",
        "values": "{\"unit\":\"cm\",\"min\":0,\"max\":600,\"scale\":0,\"step\":1}"
      },
      {
        "code": "motionless_sensitivity",
        "type": "Integer",
        "values": "{\"unit\":\"x\",\"min\":0,\"max\":10,\"scale\":0,\"step\":1}"
      },
      {
        "code": "illuminance_value",
        "type": "Integer",
        "values": "{\"unit\":\"lux\",\"min\":0,\"max\":6000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "indicator",
        "type": "Boolean",
        "values": "{}"
      }
    ]
  },
  "data_points": [
    {
      "code": "presence_state",
      "custom_name": "",
      "dp_id": 1,
      "time": 1728453923134,
      "type": "enum",
      "value": "none"
    },
    {
      "code": "sensitivity",
      "custom_name": "",
      "dp_id": 2,
      "time": 1727293526744,
      "type": "value",
      "value": 6
    },
    {
      "code": "near_detection",
      "custom_name": "",
      "dp_id": 3,
      "time": 1727293524922,
      "type": "value",
      "value": 0
    },
    {
      "code": "far_detection",
      "custom_name": "",
      "dp_id": 4,
      "time": 1727293524809,
      "type": "value",
      "value": 500
    },
    {
      "code": "human_motion_state",
      "custom_name": "",
      "dp_id": 101,
      "time": 1728453923236,
      "type": "enum",
      "value": "none"
    },
    {
      "code": "presence_time",
      "custom_name": "",
      "dp_id": 102,
      "time": 1727293524290,
      "type": "value",
      "value": 120
    },
    {
      "code": "dp_false_alarm",
      "custom_name": "",
      "dp_id": 103,
      "time": 1727293527465,
      "type": "value",
      "value": 1
    },
    {
      "code": "motionless_far_detection",
      "custom_name": "",
      "dp_id": 104,
      "time": 1727293525023,
      "type": "value",
      "value": 500
    },
    {
      "code": "motionless_sensitivity",
      "custom_name": "",
      "dp_id": 105,
      "time": 1727293526833,
      "type": "value",
      "value": 7
    },
    {
      "code": "illuminance_value",
      "custom_name": "",
      "dp_id": 106,
      "time": 1728453599989,
      "type": "value",
      "value": 0
    },
    {
      "code": "indicator",
      "custom_name": "",
      "dp_id": 107,
      "time": 1727293526948,
      "type": "bool",
      "value": true
    },
    {
      "code": "breathe_detection",
      "custom_name": "",
      "dp_id": 108,
      "time": 1727293527154,
      "type": "value",
      "value": 600
    },
    {
      "code": "breathe_sensitivity",
      "custom_name": "",
      "dp_id": 109,
      "time": 1727293527358,
      "type": "value",
      "value": 8
    },
    {
      "code": "micro_min_detection",
      "custom_name": "",
      "dp_id": 110,
      "time": 1727293525109,
      "type": "value",
      "value": 0
    },
    {
      "code": "bre_min_detection",
      "custom_name": "",
      "dp_id": 111,
      "time": 1727293527255,
      "type": "value",
      "value": 0
    },
    {
      "code": "reset_setting",
      "custom_name": "",
      "dp_id": 112,
      "time": 1711464027997,
      "type": "bool",
      "value": false
    },
    {
      "code": "false_breath",
      "custom_name": "",
      "dp_id": 113,
      "time": 1727293527583,
      "type": "bool",
      "value": true
    },
    {
      "code": "study_timer",
      "custom_name": "",
      "dp_id": 114,
      "time": 1711464027997,
      "type": "value",
      "value": 0
    },
    {
      "code": "alarm_time",
      "custom_name": "",
      "dp_id": 115,
      "time": 1727293527054,
      "type": "value",
      "value": 2
    },
    {
      "code": "alarm_volume",
      "custom_name": "",
      "dp_id": 116,
      "time": 1727293527139,
      "type": "enum",
      "value": "high"
    },
    {
      "code": "host_mode",
      "custom_name": "",
      "dp_id": 117,
      "time": 1727293524085,
      "type": "enum",
      "value": "disarm"
    },
    {
      "code": "mov_auto",
      "custom_name": "",
      "dp_id": 118,
      "time": 1727293527889,
      "type": "bool",
      "value": false
    },
    {
      "code": "micr_auto",
      "custom_name": "",
      "dp_id": 119,
      "time": 1727293527890,
      "type": "bool",
      "value": false
    },
    {
      "code": "bre_auto",
      "custom_name": "",
      "dp_id": 120,
      "time": 1727293527666,
      "type": "bool",
      "value": false
    },
    {
      "code": "batttery",
      "custom_name": "",
      "dp_id": 121,
      "time": 1711464027997,
      "type": "value",
      "value": -1
    }
  ]
}
```

**Comment by bobvandevijver** [comment]
You have not submitted the requested app diagnostics log, but I really doubt you have added it using the "Human Presence Sensor" driver as with the specification of this device you should have a alarm capability (included flowcards) and three settings.

Please remove the device and add it (again) using the "Human Presence Sensor" driver. Afterwards, if the issue is still there, you can generate the app diagnostics report and share its id here.

**Comment by pinpoint90** [comment]
![1](https://github.com/user-attachments/assets/3add7b5d-37b6-49e5-b629-5e90891b47e3)
![2](https://github.com/user-attachments/assets/c7c0468e-a660-49fb-ac2a-3f53fc6f7086)
![3](https://github.com/user-attachments/assets/cbbcef0b-9cae-485e-9c33-1e1dce4ffedb)
![4](https://github.com/user-attachments/assets/53558b2d-6313-4854-aeea-6ca2411aa8c4)

Diagnostics ID
ee8b0c5d-899d-48db-b586-db8a94f0a11e

**Comment by bobvandevijver** [comment]
How about the settings (Sensitivity, Nearest detection distance and Furthest detection distance). Do you have those?

**Comment by pinpoint90** [comment]
![Skjermbilde 2024-10-10 121752](https://github.com/user-attachments/assets/a81d1ced-6ba4-4fca-8a7e-ff9319596618)

**Comment by bobvandevijver** [comment]
This is very, very weird. Can you please (temporarily) share your device with me so I can look into this a bit more? You can send the invite to `homey-tuya[at]drenso.nl`.

**Comment by bobvandevijver** [comment]
Oh man, this was a nasty one. Thank you for sharing the device, I have fixed it and am preparing a 1.4.1 release to make it available as soon as possible.
